File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.py
def solution(my_str, n):
    answer = []
    N = len(my_str)
    idx = 0
    while idx < N:
        tmp = ''
        for _ in range(n):
            if idx < N:
                tmp += my_str[idx]
                idx += 1
        answer.append(tmp)
    return answer

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
import java.util.ArrayList;
class Solution {
    public String[] solution(String my_str, int n) {
        String[] answer = {};
        int idx = 0;
        int N = my_str.length();
        ArrayList<String> arr = new ArrayList<>();
        
        while (idx < N) {
            String tmp = "";
            for (int i = 0; i < n; i++) {
                if (idx < N) {
                    tmp += my_str.charAt(idx);
                    idx++;
                }
            }
            arr.add(tmp);
        }
        
        answer = new String[arr.size()];
        for (int i = 0; i < arr.size(); i++) {
            answer[i] = arr.get(i);
        }
        return answer;
    }
}